tests/benchmark/test_bloatnet.py

```diff
@@ -0,0 +1,382 @@
+            storage=initial_storage,  # type: ignore
+        )
+        tx = Transaction(
+            to=contract_address,
+            gas_limit=tx_gas_limit,
+            data=calldata,
+            sender=pre.fund_eoa(),
+        )
+        all_txs.append(tx)
+
+        actual_intrinsic_consumed = intrinsic_gas_calc(
+            calldata=calldata, return_cost_deducted_prior_execution=True
+        )
+
+        tx_gas_used = actual_intrinsic_consumed + tx_opcode_gas
+        total_block_gas_used += tx_gas_used
+
+        expected_storage_state[contract_address] = Account(
+            storage=Storage(
+                {
+                    HashInt(slot): HashInt(final_storage_value)
+                    for slot in range(total_sstores, total_sstores + tx_sstores_count)
+                }
+            )
+        )
+
+        total_sstores += tx_sstores_count
+
+    blockchain_test(
+        pre=pre,
+        blocks=[Block(txs=all_txs)],
+        post=expected_storage_state,
+        expected_benchmark_gas_used=total_block_gas_used,
+    )
+
+
+# Warm reads are very cheap, which means you can really fill a block
+# with them. Only fill the block by a factor of SPEEDUP.
+SPEEDUP: int = 100
+
+
+@pytest.mark.valid_from("Prague")
+def test_bloatnet_sload_warm(
+    blockchain_test: BlockchainTestFiller, pre: Alloc, fork: Fork, gas_benchmark_value: int
+):
+    """Test that loads warm storage locations many times."""
+    gas_costs = fork.gas_costs()
+    intrinsic_gas_calc = fork.transaction_intrinsic_cost_calculator()
+    tx_gas_cap = fork.transaction_gas_limit_cap() or gas_benchmark_value
+
+    # Pre-fill storage with values
+    num_slots = 100  # Number of storage slots to warm up
+    storage = Storage({HashInt(i): HashInt(0xDEADBEEF + i) for i in range(num_slots)})
+
+    # Gas costs for operations
+    # PUSH + SLOAD + POP for cold access
+    cold_sload_cost = gas_costs.G_VERY_LOW + gas_costs.G_COLD_SLOAD + gas_costs.G_BASE
+    # PUSH + SLOAD + POP for warm access
+    warm_sload_cost = gas_costs.G_VERY_LOW + gas_costs.G_WARM_SLOAD + gas_costs.G_BASE
+
+    # Calculate how many operations fit in a single transaction
+    intrinsic_gas = intrinsic_gas_calc(calldata=b"")
+    opcode_gas_budget = tx_gas_cap - intrinsic_gas
+
+    # Calculate max warm loads that can fit in a transaction
+    # (applying SPEEDUP factor to reduce the number of operations)
+    num_warm_loads_per_tx = opcode_gas_budget // (SPEEDUP * warm_sload_cost)
+
+    # Build the code that does warm loads
+    # Since storage slots are already warmed within each transaction,
+    # we just need to access them repeatedly
+    sload_code = Bytecode()
+    # First warm up all slots (these will be cold on first access in each tx)
+    for i in range(num_slots):
+        sload_code = sload_code + Op.SLOAD(i) + Op.POP
+    # Then do warm loads
+    for i in range(num_warm_loads_per_tx):
+        sload_code = sload_code + Op.SLOAD(i % num_slots) + Op.POP
```

Comment:
We could refactor to the following:

```suggestion
sload_code = Bytecode()
sload_code += sum(Op.POP(Op.SLOAD(i)) for i in range(num_slots))
sload_code += sum(Op.POP(Op.SLOAD(i % num_slots)) for i in range(num_warm_loads_per_tx))
```